test/unit/test_http_utils.py

```diff
@@ -125,26 +126,59 @@ class TestSomewhatBufferedHTTPConnection(unittest.TestCase):
     @mock.patch('httplib.HTTPResponse.__init__')
     def test_discard_buffering_arg_on_py26(
             self, mock_http_response_init):
+        with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)) \
+                as sock:
 
-        kwargs = dict(
-            sock=None, debuglevel=0, strict=0, method=None, buffering=False)
+            kwargs = dict(
+                sock=sock, debuglevel=0, strict=0, method=None, buffering=False)
 
-        response_class = \
-            swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
-        response_obj = response_class(**kwargs)
+            response_class = \
+                swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
+            response_obj = response_class(**kwargs)
 
-        if sys.version_info < (2, 7):
-            del kwargs['buffering']
+            if sys.version_info < (2, 7):
+                del kwargs['buffering']
 
-        mock_http_response_init.assert_called_once_with(
-            response_obj, **kwargs)
+            mock_http_response_init.assert_called_once_with(
+                response_obj, **kwargs)
 
-    @mock.patch('httplib.HTTPResponse.__init__', mock.Mock())
     def test_fileno_attr(self):
-        response_class = \
-            swift_scality_backend.http_utils.SomewhatBufferedHTTPConnection.HTTPResponse
-        response_obj = response_class(sock=None)
         with contextlib.closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)) \
                 as sock:
-            response_obj.fp = sock
+            response_class = \
```

Comment:
Nicolas meant: move the response_class variable outside the context manager. It's good practice to have the minimum amount of code running in a context manager (it's like having the minimum amount of code in a try/except block).
L148 is more or less a no-op, it can be moved outside